docs/markdowns/filesystem.md

```diff
@@ -1,37 +1,82 @@
+is in development.
 
-## [Filesystem (fsspec)](https://filesystem-spec.readthedocs.io/en/latest/usage.html)
+## Using URLs
 
-* [Copy conventions](https://filesystem-spec.readthedocs.io/en/latest/copying.html)
+Xet URLs should be of the form `xet://<repo_user>/<repo_name>/<branch>/<path-to-file>`, where `<path-to-file>` is optional if the URL 
+refers to a repository. The xet:// prefix is inferred as needed, or if the URL is given as https://.  
 
-The `pyxet.repo` returns a repository-filesystem that can be used with fsspec.   
-It works both like a normal filesystem and git.
+Setting a user and token is required for private repositories. They may be provided as explicit arguments to most pyxet functions, 
+or they can be passed in with the URL by prefixing `xet://<user>[:token]@xethub.com/`. For example, 
+`xet://user1:mytokenxyz@xethub.com/data_user/data_repo/main/data/survey.csv` would access the file `data/survey.csv` on 
+the branch `main` of the repo `data_user/data_repo`  with credentials `user=user1` and `token=mytokenxyz`. 
 
-```python
-import pyxet
+For example, to refer to the results.csv file in the main branch of the XetHub Flickr30k repo, the following work: 
+- xet://xethub.com/XetHub/Flickr30k/main/results.csv (all fsspec compatible packages)
+- /XetHub/Flickr30k/main/results.csv (pyxet.open) 
+- https://xethub.com/XetHub/Flickr30k/main/results.csv (pyxet.open) 
+
+## pyxet.open
+
+To open a file from an XetHub repository, you can use the `pyxet.open()` function, which takes a file URL in the format 
+`xet://<repo_user>/<repo_name>/<branch>/<path-to-file>`.
+
+Example usage of `pyxet.open`:
+```sh
+  import pyxet
+
+  # Open a file from a public repository.
+  f = pyxet.open('xet://xethub.com/XetHub/Flickr30k/main/results.csv')
+
+  # Read the contents of the file.
+  contents = f.read()
+  f.close()
+```
+
+## pyxet.XetFS
 
-fs = pyxet.XetFS("username/repo/branch", login=..., **kwargs)
+To work with a XetHub repository as a file system, you can use the `pyxet.XetFS` class. This class provides a file system handle 
+for a XetHub repository, allowing you to perform read-only operations like ls, glob, and open. The initialization of this class 
+requires a repository URL and optional arguments for branch, user, and token. 
 
-fs.ls("path/to/dir")
-fs.glob("path/to/dir/*")
+Example usage of `pyxet.XetFS`:
 
-# Files
-with fs.open("path/data.csv", 'b') as f:
-    f.read()
+```sh
+  import pyxet
 
-    ...
+  # Create a file system handle for a public repository.
+  fs = pyxet.XetFS('xet://xethub.com/XetHub/Flickr30k/main/')
 
-# Write is coming soon
-with fs.commit("message"):
-    with fs.open("path/file.txt", 'w') as f:
-        f.write("Hello, world!")
+  # List files in the repository.
+  files = fs.ls('/')
+
+  # Open a file from the repository.
+  f = fs.open('results.csv')
+
+  # Read the contents of the file.
+  contents = f.read()
+```
+
+## fsspec integration
+
```

Comment:
Let's add the fsspec link here:

[Filesystem (fsspec)](https://filesystem-spec.readthedocs.io/en/latest/usage.html)

I see other packages doing this so people interested in the spec can follow it.